modules/imgcodecs/src/loadsave.cpp

```diff
@@ -591,32 +591,47 @@ bool imreadmulti(const String& filename, std::vector<Mat>& mats, int flags)
     return imreadmulti_(filename, flags, mats);
 }
 
-static bool imwrite_( const String& filename, const Mat& image,
+static bool imwrite_( const String& filename, const std::vector<Mat>& img_vec,
                       const std::vector<int>& params, bool flipv )
 {
+    bool isMultiImg = img_vec.size() > 1;
+    std::vector<Mat> write_vec;
     Mat temp;
-    const Mat* pimage = &image;
-
-    CV_Assert( image.channels() == 1 || image.channels() == 3 || image.channels() == 4 );
+    const Mat* pimage = &img_vec[0]; //to suppress compiler warning
```

Comment:
Maybe to suppress the warning it would be better to move this variable into the following `for` loop?